core/embed/boardloader/main.c

```diff
@@ -276,6 +276,16 @@ int main(void) {
   display_init();
   display_clear();
 
+#ifdef STM32U5
+  const image_header *old_hdr = read_image_header(
+      (const uint8_t *)BOOTLOADER_START, BOOTLOADER_IMAGE_MAGIC,
+      flash_area_get_size(&BOOTLOADER_AREA));
+  secret_ensure_initialized();
```

Comment:
istm a function "secret - ensure initialized" should be called once as part of the init sequence, and then not touched again (and get/fill could then assume that someone else had already called it)

is there something i'm missing? the old discussion refered to "still halting after reconfiguring the sector", i don't see that now?

Reply:
That referred to the check done in bootloader in `secret_bhk_provision`, which is now moved to bootloader's main: https://github.com/trezor/trezor-firmware/commit/e9f84176037ef83e7fefee99e2c20586e8dd6926

so the "secret - ensure initialized" is called in both boardloader main and bootloader main, which is kinda redundant but should not break anything.